app/renderer/src/components/Inspector/Source.jsx

```diff
@@ -20,27 +52,70 @@ const Source = (props) => {
     methodCallInProgress,
     mjpegScreenshotUrl,
     isSourceRefreshOn,
+    pageSourceSearchText,
     t,
   } = props;
 
+  useEffect(() => {
+    if (!treeData || !pageSourceSearchText) {
+      return;
+    }
+
+    const nodesMatchingSearchTerm = [];
+
+    /**
+     * If any search text is entered, we will try to find matching nodes in the tree.
+     * and expand their parents to make the nodes visible that matches the
+     * search text.
+     *
+     * hierarchy is an array of node keys representing the path from the root to the
+     * current node.
+     */
+    const findNodesToExpand = (node, hierarchy) => {
+      /* Node title will an object representing a react element.
+       * renderToString method will construct a HTML DOM string
+       * which can be used to match against the search text.
+       *
+       * If any node that matches the search text is found, we will add all its
+       * parents to the 'nodesMatchingSearchTerm' array to make them automatically expand.
+       */
+      const nodeText = renderToString(node.title).toLowerCase();
+      if (nodeText.includes(pageSourceSearchText.toLowerCase())) {
+        nodesMatchingSearchTerm.push(...hierarchy);
+      }
+      if (node.children) {
+        node.children.forEach((c) => findNodesToExpand(c, [...hierarchy, node.key]));
+      }
+    };
+    treeData.forEach((node) => findNodesToExpand(node, [node.key]));
+    setExpandedPaths(uniq(nodesMatchingSearchTerm));
+  }, [treeData, pageSourceSearchText]);
```

Comment:
I wanted to test this but unfortunately I got an error: `Cannot access 'treeData' before initialization`.
If I moved the `useEffect` hook below the `treeData` definition, then I got another error: `Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.`
I checked and this happens due to too many updates in `treeData`.

Reply:
Got it. It appears that `treeData` isn't necessary as a dependency for the hook. Instead, we just need to listen for changes in the search input and re-render the tree as needed.

Interestingly, this issue didn't come up when the PR was initially created